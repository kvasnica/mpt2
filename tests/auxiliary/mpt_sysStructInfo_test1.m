function mpt_sysStructInfo_test1

% tests whether we display information about LTI/PWA systems correctly:
% http://control.ee.ethz.ch/~mpt/hg/mpt?cs=84289275a45c

% LTI systems
Double_Integrator
mpt_sysStructInfo(sysStruct)

% PWA systems
two_tanks
mpt_sysStructInfo(sysStruct)

% PWA system - 3D
pwa3d
mpt_sysStructInfo(sysStruct)
