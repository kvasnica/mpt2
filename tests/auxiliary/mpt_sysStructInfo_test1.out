LTI system, 2 states, 1 input, 2 outputs


PWA system (4 dynamics), 2 states, 2 inputs, 1 output

Guards of dynamics 1:
x(1) - x(2) >= 0
x(1) >= 0.3
u(2) >= 0.5

Guards of dynamics 2:
x(1) - x(2) >= 0
x(1) <= 0.3
u(2) >= 0.5

Guards of dynamics 3:
x(1) - x(2) >= 0
x(1) >= 0.3
u(2) <= 0.5

Guards of dynamics 4:
x(1) - x(2) >= 0
x(1) <= 0.3
u(2) <= 0.5


PWA system (2 dynamics), 3 states, 1 input, 3 outputs

Guards of dynamics 1:
x(2) <= 1

Guards of dynamics 2:
x(2) >= 1
